components/brave_rewards/browser/rewards_service_browsertest.cc

```diff
@@ -1409,3 +1490,143 @@ IN_PROC_BROWSER_TEST_F(BraveRewardsBrowserTest,
   // Stop observing the Rewards service
   rewards_service_->RemoveObserver(this);
 }
+
+
+IN_PROC_BROWSER_TEST_F(BraveRewardsBrowserTest,
+    InsufficientNotificationForVerifiedsZeroAmountZeroPublishers) {
+  rewards_service_->GetNotificationService()->AddObserver(this);
+  EnableRewards();
+  CheckInsufficientFundsForTesting();
+  WaitForInsufficientFundsNotification();
+  const brave_rewards::RewardsNotificationService::RewardsNotificationsMap&
+      notifications = rewards_service_->GetAllNotifications();
+
+  if (notifications.empty()) {
+    SUCCEED();
+  }
+  bool notification_shown = false;
+  for (const auto& notification : notifications) {
+    if (notification.second.type_ ==
+        brave_rewards::RewardsNotificationService::RewardsNotificationType
+        ::REWARDS_NOTIFICATION_INSUFFICIENT_FUNDS) {
+      notification_shown = true;
+      break;
+    }
+  }
+  EXPECT_FALSE(notification_shown);
+}
+
+IN_PROC_BROWSER_TEST_F(BraveRewardsBrowserTest,
+                       InsufficientNotificationForVerifiedsDefaultAmount) {
+  rewards_service_->AddObserver(this);
+  rewards_service_->GetNotificationService()->AddObserver(this);
+  EnableRewards();
+  // Claim grant using panel
+  const bool use_panel = true;
+  ClaimGrant(use_panel);
+
+  // Visit publishers
+  const bool verified = true;
+  while (!last_publisher_added_) {
+    VisitPublisher("duckduckgo.com", verified);
+    VisitPublisher("bumpsmack.com", verified, 1);
+    VisitPublisher("google.com", !verified, 2, true);
+  }
+
+  CheckInsufficientFundsForTesting();
+  WaitForInsufficientFundsNotification();
+  const brave_rewards::RewardsNotificationService::RewardsNotificationsMap&
+      notifications = rewards_service_->GetAllNotifications();
+
+  if (notifications.empty()) {
+    SUCCEED();
+  }
+  bool notification_shown = false;
+  for (const auto& notification : notifications) {
+    if (notification.second.type_ ==
+        brave_rewards::RewardsNotificationService::RewardsNotificationType
+        ::REWARDS_NOTIFICATION_INSUFFICIENT_FUNDS) {
+      notification_shown = true;
+      break;
+    }
+  }
+  EXPECT_FALSE(notification_shown);
+}
+
+IN_PROC_BROWSER_TEST_F(BraveRewardsBrowserTest,
+                       InsufficientNotificationForVerifiedsSufficientAmount) {
+  rewards_service_->AddObserver(this);
+  rewards_service_->GetNotificationService()->AddObserver(this);
+
+  EnableRewards();
+  // Claim grant using panel
+  const bool use_panel = true;
+  ClaimGrant(use_panel);
+
+  // Visit publishers
+  const bool verified = true;
+  while (!last_publisher_added_) {
+    VisitPublisher("duckduckgo.com", verified);
+    VisitPublisher("bumpsmack.com", verified, 1);
+    VisitPublisher("google.com", !verified, 2, true);
+  }
+
+  rewards_service_->SetContributionAmount(50.0);
```

Comment:
if we set amount to 50 and claim grant which is 30, then notification should be shown right?

Reply:
This was an oversight, this test also passed because of the above `VisitPublisher` not counting the last publisher in. This is fixed.